Dockerfile

```diff
@@ -0,0 +1,68 @@
+# ---------- 1st stage ----------
+# The first stage adds the necessary libraries to build native add-ons (eg. bcrypt) and then installs
+# the server dependencies
+FROM node:12-alpine as builder
+
+# By default, `alpine` images don't have necessary tools to install native add-ons, so we use the
+# multistage build to install the necessary tools, and build the dependencies which will then be
+# copied over to the next stage (this stage will be discarded, along with the installed tools, to
+# make the image lighter).
+RUN apk add --no-cache python make g++ git
+
+# Ensure that we don't spend time installing dependencies that are only needed for the client
+# application.
+ENV SB_SERVER_ONLY=1
+
+# Set the working directory to where we want to install the dependencies; this directory doesn't
+# really matter as it will not be present in the final image, but keeping the path short makes
+# copying from it easier in the next stage.
+WORKDIR /shieldbattery
+
+# Copy the whole repository to the image, *except* the stuff marked in the `.dockerignore` file
+COPY . .
+
+# Install only the server and main dependencies (`app` dependencies should be built into the
+# Electron app) and install only production dependencies to make the image smaller.
+RUN cd server && yarn && cd .. && yarn
```

Comment:
Shouldn't the `ENV NODE_ENV=production` be declared above this line if we only want to install the production dependencies?

EDIT: I see we prune the server deps to only the production ones below. Maybe a comment outlining why that's necessary would help here.

Reply:
Yeah, we need the non-production deps so that we can use webpack, then we prune it back out after doing so. I'll add a comment